pkg/agent/cniserver/pod_configuration_windows.go

```diff
@@ -0,0 +1,114 @@
+		return fmt.Errorf("container lock cannot be null")
+	}
+	ovsPortName := ifConfig.InterfaceName
+	expectedEp, ok := pc.ifConfigurator.getEndpoint(ovsPortName)
+	if !ok {
+		return fmt.Errorf("failed to find HNSEndpoint %s", ovsPortName)
+	}
+	hostIfAlias := fmt.Sprintf("%s (%s)", util.ContainerVNICPrefix, ovsPortName)
+	containerID := ifConfig.ContainerID
+	go func() {
+		klog.Infof("Waiting for interface %s to be created", hostIfAlias)
+		err := wait.PollImmediate(time.Second, 60*time.Second, func() (bool, error) {
+			curEp, ok := pc.ifConfigurator.getEndpoint(ovsPortName)
+			if !ok {
+				return true, fmt.Errorf("failed to find HNSEndpoint %s", ovsPortName)
+			}
+			if curEp.Id != expectedEp.Id {
+				klog.Warningf("detected HNEEndpoint change for port %s, exit current thread", ovsPortName)
+				return true, nil
+			}
+			if !util.HostInterfaceExists(hostIfAlias) {
+				klog.Infof("Waiting for interface %s to be created", hostIfAlias)
+				return false, nil
+			}
+			if err := pc.connectInterfaceToOVSCommon(ovsPortName, ifConfig); err != nil {
+				return true, fmt.Errorf("failed to connect to ovs for container %s: %v", containerID, err)
+			}
+			return true, nil
+		})
+		if err != nil {
+			klog.Errorf("failed to create OVS port for container %s: %v", containerID, err)
+		}
+	}()
+	return nil
+}
+
+// connectInterfaceToOVS connects an existing interface to ovs br-int.
+func (pc *podConfigurator) connectInterfaceToOVS(
+	podName string,
+	podNameSpace string,
+	containerID string,
+	hostIface *current.Interface,
+	containerIface *current.Interface,
+	ips []*current.IPConfig,
+	containerAccess *containerAccessArbitrator,
+) (*interfacestore.InterfaceConfig, error) {
+	// Use the outer veth interface name as the OVS port name.
+	ovsPortName := hostIface.Name
+	containerConfig := buildContainerConfig(ovsPortName, containerID, podName, podNameSpace, containerIface, ips)
+	hostIfAlias := fmt.Sprintf("%s (%s)", util.ContainerVNICPrefix, ovsPortName)
+	// - For ContainerD runtime, the container interface is created after CNI replying the network setup result.
+	//   So for such case we need to use asynchronous way to wait for interface to be created.
+	// - For Docker runtime, antrea-agent still creates OVS port synchronously.
+	// - Here antrea-agent determines the way of OVS port creation by checking if container interface is yet created.
+	//   If one day ContainerD runtime changes the behavior and container interface can be created when attaching
+	//   HNSEndpoint/HostComputeEndpoint. The current implementation sill works. It will choose the synchronized
+	//   way to create OVS port.
+	if util.HostInterfaceExists(hostIfAlias) {
+		return containerConfig, pc.connectInterfaceToOVSCommon(ovsPortName, containerConfig)
+	} else {
+		return containerConfig, pc.connectInterfaceToOVSAsync(containerConfig, containerAccess)
+	}
+}
+
+func (pc *podConfigurator) reconcileMissingPods(pods sets.String, containerAccess *containerAccessArbitrator) error {
```

Comment:
there is no point in returning an `error` object if 1) you ignore it at the caller, 2) you never return an error but instead log everything directly in the function

Reply:
Exactly, remove the return value.